### PR TITLE
Fix: refactor of ESolver_KS_PW::init_after_vc with new psi initializer

### PR DESCRIPTION
**Pull request by kirk0830**
codes and annotations

### CODE REVIEW

**Comment by kirk0830** [comment]
image

**Review by hongriTianqi — APPROVED** [review]

File: source/module_psi/test/psi_initializer_integrated_test.cpp [hunk 1]
```diff
@@ -167,6 +167,121 @@ TEST_F(IntegratedInitializerTest, CalPsiGAtomic) {
...
 }
 
+TEST_F(IntegratedInitializerTest, CalPsiGAtomicRandom) {
+
+    int error = 0;
+    std::string line_new;
+    std::smatch match_new;
+    // compare wavefunctions
+    // init_wfc = "atomic+random"
+    error = std::system("cp ../../../../source/module_psi/test/support/atomic_random_new ./INPUT");
```
Comment (hongriTianqi) [inline source/module_psi/test/psi_initializer_integrated_test.cpp]: We can use cmake install to find support files

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: source/module_psi/test/psi_initializer_integrated_test.cpp [hunk 2]
```diff
@@ -167,6 +167,121 @@ TEST_F(IntegratedInitializerTest, CalPsiGAtomic) {
...
+    // init_wfc = "atomic"
+    error = std::system("cp ../../../../source/module_psi/test/support/atomic_new_soc ./INPUT");
+    error = std::system("rm ./STRU");
+    error = std::system("cp ../../../../source/module_psi/test/support/STRU_soc ./STRU");
+    error = std::system("cp ../../../../tests/PP_ORB/As_dojo_soc.upf ./As_dojo_soc.upf");
+    error = std::system("cp ../../../../tests/PP_ORB/Ga_dojo_soc.upf ./Ga_dojo_soc.upf");
+    error = std::system("cp ../../../../tests/PP_ORB/As_gga_8au_100Ry_2s2p1d.orb ./As_gga_8au_100Ry_2s2p1d.orb");
+    error = std::system("cp ../../../../tests/PP_ORB/Ga_gga_9au_60Ry_2s2p2d.orb ./Ga_gga_9au_60Ry_2s2p2d.orb");
+
+    error = std::system("../../../abacus");
```
Comment (hongriTianqi) [inline source/module_psi/test/psi_initializer_integrated_test.cpp]: Is this good practice to call abacus in unit test?

File: source/module_psi/test/psi_initializer_integrated_test.cpp [hunk 3]
```diff
@@ -123,7 +123,7 @@ TEST_F(IntegratedInitializerTest, CalPsiGAtomic) {
     std::smatch match_new;
     // compare wavefunctions
     // init_wfc = "atomic"
-    error = std::system("cp ../../../../source/module_psi/test/support/atomic_new ./INPUT");
+    error = std::system("cp ./support/atomic_new ./INPUT");
     error = std::system("../../../abacus");
```
Comment (hongriTianqi) [inline source/module_psi/test/psi_initializer_integrated_test.cpp]: Please avoid using the "abacus" executable in unit test.

**Review by hongriTianqi — CHANGES_REQUESTED** [review]

File: docs/advanced/input_files/input-main.md [hunk 4]
```diff
@@ -463,16 +464,36 @@ These variables are used to control general system parameters.
...
   Available options are:
+  
+  `psi_initializer 0` case:
   - atomic: from atomic pseudo wave functions. If they are not enough, other wave functions are initialized with random numbers.
   - atomic+random: add small random numbers on atomic pseudo-wavefunctions
   - file: from file
   - random: random numbers
+  
+  `psi_initializer 1` case:
+  - atomic: from atomic pseudo wave functions. If they are not enough, other wave functions are initialized with random numbers.
```
Comment (hongriTianqi) [inline docs/advanced/input_files/input-main.md]: It is better not to repeat information on "atomic" and "atom+random" choices?

File: docs/advanced/input_files/input-main.md [hunk 5]
```diff
@@ -463,16 +464,36 @@ These variables are used to control general system parameters.
...
+  - atomic+random: add small random numbers on atomic pseudo-wavefunctions
+  - nao: from numerical atomic orbitals. If they are not enough, other wave functions are initialized with random numbers.
```
Comment (hongriTianqi) [inline docs/advanced/input_files/input-main.md:491]:
```suggestion
  - nao: from numerical atomic orbitals. If they are not enough, other wave functions are initialized with random numbers. Available only if `psi_initializer` is set to 1.

File: docs/advanced/input_files/input-main.md [hunk 6]
```diff
@@ -463,16 +464,36 @@ These variables are used to control general system parameters.
...
+  - nao+random: add small random numbers on numerical atomic orbitals
```
Comment (hongriTianqi) [inline docs/advanced/input_files/input-main.md:492]:
```suggestion
  - nao+random: add small random numbers on numerical atomic orbitals. Available only if `psi_initializer` is set to 1.

**Review by hongriTianqi — APPROVED** [review]

**Pull request merged**